[io][ML] Sporadic crash of the RDataLoader (and RBatchGenerator)

### Check duplicate issues.

- [x] Checked for duplicates

### Description

The test `pyunittests-bindings-pyroot-pythonizations-ml-dataloader` sometimes fail with the error:
```
2026-03-25T14:52:48.8191352Z ======================================================================
2026-03-25T14:52:48.8191749Z FAIL: test14_big_data_replacement_true (ml_dataloader.DataLoaderRandomUndersampling.test14_big_data_replacement_true)
2026-03-25T14:52:48.8192161Z ----------------------------------------------------------------------
2026-03-25T14:52:48.8192395Z Traceback (most recent call last):
2026-03-25T14:52:48.8192801Z   File "/github/home/ROOT-CI/src/bindings/pyroot/pythonizations/test/ml_dataloader.py", line 4837, in test14_big_data_replacement_true
2026-03-25T14:52:48.8193296Z     test(batch_size, entries_in_rdf_major, entries_in_rdf_minor, sampling_ratio)
2026-03-25T14:52:48.8193721Z   File "/github/home/ROOT-CI/src/bindings/pyroot/pythonizations/test/ml_dataloader.py", line 4823, in test
2026-03-25T14:52:48.8194106Z     self.assertLess(len(set(flat_z_train)), len(flat_z_train))
2026-03-25T14:52:48.8194346Z AssertionError: 20092 not less than 20092
```

### Reproducer

`pyunittests-bindings-pyroot-pythonizations-ml-dataloader`

### ROOT version

master

### Installation method

all

### Operating system

all

### Additional context

_No response_

--- a/bindings/pyroot/pythonizations/test/ml_dataloader.py
+++ b/bindings/pyroot/pythonizations/test/ml_dataloader.py
@@ -4725,7 +4725,7 @@ class DataLoaderRandomUndersampling(unittest.TestCase):
 
         # max samling strategy to guarantee duplicate sampled entires
         max_sampling_ratio = entries_in_rdf_minor / entries_in_rdf_major
-        sampling_ratio = round(uniform(0.01, max_sampling_ratio), 2)
+        sampling_ratio = round(uniform(0.01, max_sampling_ratio - 0.01), 2)
 
         error_message = f"\n Batch size: {batch_size}\
             Number of major entries: {entries_in_rdf_major} \

